src/libzstd.jl

```diff
@@ -54,26 +60,26 @@ function reset!(cstream::CStream, srcsize::Integer)
         # explicitly specified.
         srcsize = ZSTD_CONTENTSIZE_UNKNOWN
     end
-    return LibZstd.ZSTD_CCtx_setPledgedSrcSize(cstream.ptr, srcsize)
+    return LibZstd.ZSTD_CCtx_setPledgedSrcSize(cstream, srcsize)
     #return ccall((:ZSTD_resetCStream, libzstd), Csize_t, (Ptr{Cvoid}, Culonglong), cstream.ptr, srcsize)
 
 end
 
 function compress!(cstream::CStream)
-    return LibZstd.ZSTD_compressStream(cstream.ptr, pointer_from_objref(cstream.obuffer), pointer_from_objref(cstream.ibuffer))
+    return LibZstd.ZSTD_compressStream(cstream, cstream, cstream)
```

Comment:
Unless the cstream is managing the memory for the buffer in a way that’s not visible to the runtime (e.g. finalizer), it is safe to access the buffer fields here. While passing in the cstream can work, I feel like it’s a bit too confusing to read…

Reply:
Resolved in 583d955